audio/drivers/coreaudio.cpp

```diff
@@ -0,0 +1,409 @@
+//=============================================================================
+//  MusE Score
+//  Linux Music Score Editor
+//
+//  Copyright (C) 2002-2010 Werner Schweer and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "mididriver.h"
+
+#include "libmscore/score.h"
+
+#include "mscore/preferences.h"
+#include "mscore/seq.h"
+
+#include "coreaudio.h"
+
+#include <CoreAudio/CoreAudio.h>
+#include <AudioUnit/AudioUnit.h>
+#include <AudioToolbox/AudioToolbox.h>
+
+#include <libkern/OSAtomic.h>
+#include <Carbon/Carbon.h>
+
+namespace Ms {
+
+AudioUnit audioUnit = NULL;
+
+//---------------------------------------------------------
+//   CoreAudio
+//---------------------------------------------------------
+
+CoreAudio::CoreAudio(Seq* s)
+   : Driver(s)
+      {
+      _sampleRate = 48000;    // will be replaced by device default sample rate
```

Comment:
This needs to be 44100 (44.1 kHz) in order for the notes to sound at the correct pitch, at least with the default soundfont.